src/package-linker.js

```diff
@@ -403,6 +413,13 @@ export default class PackageLinker {
         topLevelDependencies,
         async ([dest, pkg]) => {
           if (pkg._reference && pkg._reference.location && pkg.bin && Object.keys(pkg.bin).length) {
+            if (
+              this._isChildOfLinkedDep(linkTypeDepsLocations, pkg._reference.location) ||
+              this._isChildOfLinkedDep(linkTypeDepsLocations, dest)
```

Comment:
Looks like we only use `_isChildOfLinkedDep` in a combination of `dest` and `pkg._reference.location` so may be we can pass it two values instead and be more efficient?